dace/transformation/dataflow/vectorization.py

```diff
@@ -18,100 +143,341 @@ class Vectorization(transformation.SingleStateTransformation):
+
+                try:
+                    for idx, expr in enumerate(subset):
+                        if isinstance(expr, tuple):
+                            for ex in expr:
+                                ex = symbolic.pystr_to_symbolic(ex)
+                                symbols = ex.free_symbols
+                                if param in symbols:
+                                    if array.strides[idx] != 1:
+                                        return False
+                        else:
+                            expr = symbolic.pystr_to_symbolic(expr)
+                            symbols = expr.free_symbols
                             if param in symbols:
-                                if array.strides[idx] == 1:
-                                    found = True
-                                else:
+                                if array.strides[idx] != 1:
                                     return False
-                    else:
-                        expr = symbolic.pystr_to_symbolic(expr)
-                        symbols = expr.free_symbols
-                        if param in symbols:
-                            if array.strides[idx] == 1:
-                                found = True
-                            else:
-                                return False
-            except TypeError:  # cannot determine truth value of Relational
-                return False
+                except TypeError:  # cannot determine truth value of Relational
+                    return False
+
+            for e in state.in_edges(node):
+                # Check for valid copies from other tasklets and/or streams
+                if e.data.data is not None:
+                    src_node = state.memlet_path(e)[0].src
+                    if not isinstance(src_node, (nodes.Tasklet, nodes.AccessNode)):
+                        # Make sure we only have Code->Code copies and from arrays
+                        return False
+
+                    if isinstance(src_node, nodes.AccessNode):
+                        src_desc = src_node.desc(sdfg)
+                        if isinstance(src_desc, data.Stream) and self.target == dtypes.ScheduleType.SVE_Map:
+                            # Stream pops are not implemented
+                            return False
+
+        # Check if it is possible to vectorize data container
+        if self.target == dtypes.ScheduleType.FPGA_Device and self._level == 0:
+            maps_to_vectorize, data_descriptors_to_vectorize = collect_maps_to_vectorize(sdfg, state, map_entry)
+
+            old_map_entry = self.map_entry
+            self._level = 1  # To prevent infinte loop
+
+            for m in maps_to_vectorize:
+                self.map_entry = m
+
+                if not self.can_be_applied(get_innermost_state_for_node(sdfg, m), expr_index,
+                                           get_innermost_sdfg_for_node(sdfg, m), permissive):
```

Comment:
This is very wasteful, why not collect this information while you're collecting the maps anyway?